topic-offset-service/src/main/java/uk/gov/gchq/palisade/service/topicoffset/service/KafkaHealthIndicator.java

```diff
@@ -0,0 +1,106 @@
+import org.springframework.boot.actuate.health.Health;
+import org.springframework.boot.actuate.health.HealthIndicator;
+import org.springframework.stereotype.Component;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+/**
+ * Kafka health indicator. Check that the consumer group can be accessed and is registered with the cluster,
+ * if not mark the service as unhealthy.
+ */
+@Component("kafka")
+@ConditionalOnEnabledHealthIndicator("kafka")
+public class KafkaHealthIndicator implements HealthIndicator {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(KafkaHealthIndicator.class);
+
+    private final String groupId;
+    private final AdminClient adminClient;
+
+    /**
+     * Requires the AdminClient to interact with Kafka
+     * *
+     *
+     * @param groupId     of the cluster
+     * @param adminClient of the cluster
+     */
+    public KafkaHealthIndicator(@Value("${akka.kafka.consumer.kafka-clients.group.id}") final String groupId, final AdminClient adminClient) {
+        this.groupId = groupId;
+        this.adminClient = adminClient;
+
+    }
+
+    @Override
+    public Health getHealth(final boolean includeDetails) {
+        if (performCheck()) {
+            return Health.up().withDetail("group", this.groupId).build();
+        } else {
+            return Health.down().withDetail("group", this.groupId).build();
+        }
+    }
+
+    /**
+     * Health endpoint
+     * @return the {@code Health} object
+     */
+    @Override
+    public Health health() {
+        if (performCheck()) {
+            return Health.up().build();
+        } else {
+            return Health.down().build();
+        }
+    }
+
+    private boolean performCheck() {
+
+        try {
+            Map<String, ConsumerGroupDescription> groupDescriptionMap = this.adminClient.describeConsumerGroups(Collections.singletonList(this.groupId))
+                    .all()
+                    .get(1, TimeUnit.SECONDS);
+
+            ConsumerGroupDescription consumerGroupDescription = groupDescriptionMap.get(this.groupId);
+
+            LOGGER.debug("Kafka consumer group ({}) state: {}", groupId, consumerGroupDescription.state());
+
+            if (consumerGroupDescription.state() == ConsumerGroupState.STABLE) {
+                return consumerGroupDescription.members().stream()
+                        .noneMatch(member -> (member.assignment() == null || member.assignment().topicPartitions().isEmpty()));
+            }
+        } catch (InterruptedException | ExecutionException | TimeoutException e) {
+            LOGGER.warn("Timeout during Kafka health check for group {}", this.groupId, e);
+            Thread.currentThread().interrupt();
+            return false;
+        }
+        return true;
+    }
```

Comment:
```suggestion
    private boolean performCheck() {
        try {
            Map<String, ConsumerGroupDescription> groupDescriptionMap = this.adminClient.describeConsumerGroups(Collections.singletonList(this.groupId))
                    .all()
                    .get(1, TimeUnit.SECONDS);

            ConsumerGroupDescription consumerGroupDescription = groupDescriptionMap.get(this.groupId);
            LOGGER.debug("Kafka consumer group ({}) state: {}", groupId, consumerGroupDescription.state());

            if (consumerGroupDescription.state() == ConsumerGroupState.STABLE) {
                boolean assignedGroupPartition = consumerGroupDescription.members().stream()
                        .noneMatch(member -> (member.assignment() == null || member.assignment().topicPartitions().isEmpty()));
                if (!assignedGroupPartition) {
                    LOGGER.error("Failed to find kafka topic-partition assignments");
                }
                return assignedGroupPartition;
            }

        } catch (InterruptedException e) {
            LOGGER.warn("Await on future interrupted", e);
            Thread.currentThread().interrupt();
        } catch (ExecutionException | TimeoutException e) {
            LOGGER.warn("Timeout connecting to kafka", e);
        }
        return false;
    }
```

Reply:
Other than some whitespace changes, the main diff here is the split out catch block at the bottom that allows for seperate error handling